From c61ed0046f361d95784041e2a6617c75d89fff58 Mon Sep 17 00:00:00 2001
From: klzgrad <kizdiv@gmail.com>
Date: Thu, 25 Jan 2018 00:11:23 -0500
Subject: [PATCH] net/socket: Allow higher limits for proxies

As an intermediary proxy we should not enforce stricter connection
limits in addition to what the user is already enforcing.
---
 src/net/socket/client_socket_pool_manager.cc | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/net/socket/client_socket_pool_manager.cc b/src/net/socket/client_socket_pool_manager.cc
index 7c7946b3b..aa1730de7 100644
--- a/src/net/socket/client_socket_pool_manager.cc
+++ b/src/net/socket/client_socket_pool_manager.cc
@@ -148,7 +148,6 @@ void ClientSocketPoolManager::set_socket_soft_cap_per_pool_for_test(
     HttpNetworkSession::SocketPoolType pool_type,
     size_t socket_count) {
   DCHECK_LT(0u, socket_count);     // At least one socket must be allowed.
-  DCHECK_GE(2048u, socket_count);  // For now, we pick a ceiling of 2^11.
   g_socket_soft_cap_per_pool[std::to_underlying(pool_type)] = socket_count;
   DCHECK_GE(g_socket_soft_cap_per_pool[std::to_underlying(pool_type)],
             g_max_sockets_per_group[std::to_underlying(pool_type)]);
@@ -165,7 +164,6 @@ void ClientSocketPoolManager::set_max_sockets_per_group_for_test(
     HttpNetworkSession::SocketPoolType pool_type,
     size_t socket_count) {
   DCHECK_LT(0u, socket_count);    // At least one socket must be allowed.
-  DCHECK_GE(512u, socket_count);  // For now, we pick a ceiling of 2^9.
   g_max_sockets_per_group[std::to_underlying(pool_type)] = socket_count;
 
   DCHECK_GE(g_socket_soft_cap_per_pool[std::to_underlying(pool_type)],
@@ -193,7 +191,6 @@ void ClientSocketPoolManager::set_max_sockets_per_proxy_chain(
   // We set out explicit limits here because they are hard coded in the
   // enterprise policy MaxConnectionsPerProxy.
   CHECK_GE(socket_count, 6u);
-  CHECK_LE(socket_count, 256u);
   // LINT.ThenChange(/net/socket/client_socket_pool_manager.cc:SetMaxConnectionsPerProxyChain)
   g_max_sockets_per_proxy_chain[std::to_underlying(pool_type)] = socket_count;
 }
-- 
2.53.0

